productivity-suite/app/fragments/news/src/root.tsx

```diff
@@ -0,0 +1,58 @@
+// @refresh reload
+import { Suspense } from "solid-js";
+import { HydrationScript, isServer } from "solid-js/web";
+import { ErrorBoundary, Routes, useServerContext, Route } from "solid-start";
+import { PageEvent } from "solid-start/server";
+import { createContext, useContext } from "solid-js";
+import "./root.css";
+import { News } from "./routes";
+
+export const ServerContext = createContext<PageEvent>({} as any);
+
+type ManifestEntry = {
+  type: "script" | "style";
+  href: string;
+};
+
+export default function Root() {
+  const context = useServerContext();
+
+  let script: ManifestEntry = null;
+  let style: ManifestEntry = null;
+
+  if (isServer) {
+    const manifestEntries = context.env.manifest[
+      "entry-client"
+    ] as ManifestEntry[];
+    script = manifestEntries.find((e) => e.type === "script");
+    style = manifestEntries.find((e) => e.type === "style");
+  }
+
+  const isSSR = import.meta.env.START_SSR;
+
+  return (
+    <div>
+      {isServer && (
+        <link rel="stylesheet" href={`/_fragment/news/public${style.href}`} />
+      )}
+
+      <ErrorBoundary>
+        <Suspense fallback={<div class="news-list-nav">Loading...</div>}>
+          <Routes>
+            <Route path="/*" component={News} />
+          </Routes>
+        </Suspense>
+      </ErrorBoundary>
+
+      {isSSR && <HydrationScript />}
+
+      {isServer && (
```

Comment:
what's the different between `isSSR` and `isServer`? 😮

Reply:
`isSSR` is true when the app is in SSR mode (as determined by the config passed into the solid vite plugin), `isServer` is true when the code is running in the worker.

Reply:
I am not sure I get it.... what's the difference? can't you just use one? 🤔

Reply:
Nope, they're different. An app is in SSR mode or not. If you have an app running in SSR mode, the code can execute on either the server (as part of the initial async rendering on the worker) or on the client (as part of hydration). Similar to routing perhaps, you can have an SSR app that has routing, but your routing code will need to check if it's in a server context to know whether to use `document.location` or the request from context.

Also FYI these aren't variables I made up. They're copied from the `solid-start` source: https://github.com/solidjs/solid-start/blob/main/packages/start/root/Scripts.tsx

Reply:
ah.... I see.... the names are pretty confusing to me, but I sort of get it now

thanks for the clarification 🙂